### PR TITLE
List adjustments

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Adds the origin of the blueprints to the list.
- Packs from Server or packs are "Local"
-

Video of change: 

ezgif-4-bb851e1255

Review please

### CODE REVIEW

File: src/main/java/com/ldtteam/structurize/client/gui/WindowSwitchPack.java [hunk 1]
```diff
@@ -244,6 +304,10 @@ private static void fillForMeta(final Pane rowPane, final StructurePackMeta pack
         {
             rowPane.findPaneOfTypeByID("icon" + side, Image.class).setImage(OutOfJarResourceLocation.of(MOD_ID, packMeta.getPath().resolve(packMeta.getIconPath())), false);
         }
+
+        //if (packMeta.getPath().)
         rowPane.findPaneOfTypeByID("select" + side, Button.class).setTextColor(ChatFormatting.BLACK.getColor());
+        //todo size is + 2*number of categories. A category makes all other stuff invisible and just has a big label.
```
Comment (Nightenom) [inline src/main/java/com/ldtteam/structurize/client/gui/WindowSwitchPack.java]: Huh?

File: src/main/java/com/ldtteam/structurize/storage/BlueprintPlacementHandling.java [hunk 2]
```diff
@@ -157,7 +157,7 @@ public static void handlePlacement(final BlueprintSyncMessage blueprintSyncMessa
                     modList.add(mod.getModId());
                 }
 
-                StructurePacks.discoverPackAtPath(blueprintParentPath, false, modList, true);
+                StructurePacks.discoverPackAtPath(blueprintParentPath, false, modList, true, "Local");
```
Comment (Nightenom) [inline src/main/java/com/ldtteam/structurize/storage/BlueprintPlacementHandling.java]: Constant or enum, plus translation, elsewhere too

File: src/main/resources/assets/structurize/gui/windowswitchpack.xml [hunk 3]
```diff
@@ -3,9 +3,13 @@
 
     <button id="cancel" source="structurize:textures/gui/buildtool/button_medium.png" label="$(structurize.gui.buildtool.cancel)" size="86 17" pos="3 3" textcolor="black" texthovercolor="gray"/>
     <input id="filter" size="129 17" pos="150 3" maxlength="25" visible="true"/>
+    <text id="packname" pos="0 0" size="119 20" color="white" textscale="1.5"/>
 
-    <list size="420 200" pos="0 20" id="packs" visible="false">
-        <box size="100% 100" linewidth="0">
+    <list size="420 200" pos="0 40" id="packs" visible="false">
+        <box size="100% 100" pos="0 20" linewidth="0">
```
Comment (Nightenom) [inline src/main/resources/assets/structurize/gui/windowswitchpack.xml:9]: Box with zero linewidth is view

File: src/main/java/com/ldtteam/structurize/client/gui/WindowSwitchPack.java [hunk 4]
```diff
@@ -202,26 +245,43 @@ public void updatePacks()
...
                 }
                 else
                 {
                     rowPane.findPaneByID("box2").hide();
                 }
+
+                if (obj1 instanceof String)
+                {
+                    rowPane.findPaneByID("box0").show();
+                    rowPane.findPaneOfTypeByID("category", Text.class).setText(Component.literal(((String) obj1).substring(0, 1).toUpperCase(Locale.US) + ((String) obj1).substring(1)));
```
Comment (Nightenom) [inline src/main/java/com/ldtteam/structurize/client/gui/WindowSwitchPack.java]: This component is sus - many substrings - ideally fnc for capitalize, also instanceof pattern for string.. and above too

**Review by Nightenom — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/structurize/gui/windowswitchpack.xml (hunk 3 above)
Comment (Raycoms) [reply src/main/resources/assets/structurize/gui/windowswitchpack.xml:9]: We always use boxes for the inner list boxing.

Comment (Nightenom) [reply src/main/resources/assets/structurize/gui/windowswitchpack.xml:9]: True, but shouldn't matter imho

**Review by someaddons — APPROVED** [review]

**Pull request merged**